pyxem/data/simulated_strain.py

```diff
@@ -0,0 +1,119 @@
+    )
+    in_frame = (
+        (transformed.data[:, 0] >= 0)
+        & (transformed.data[:, 0] < shape[1])
+        & (transformed.data[:, 1] >= 0)
+        & (transformed.data[:, 1] < shape[0])
+    )
+    spot_coords = np.round(transformed.data[in_frame]).astype(int)
+    if transformation_matrix is not None:
+        direct_beam_position = direct_beam_position + (0,)
+        spot_coords = (
+            (spot_coords - direct_beam_position) @ transformation_matrix
+        ) + direct_beam_position
+    spot_intens = transformed.intensity[in_frame]
+    pattern = np.zeros(shape)
+    # checks that we have some spots
+    if spot_intens.shape[0] == 0:
+        return pattern
+    else:
+        for cord, inten in zip(spot_coords, spot_intens):
+            rr, cc = disk(cord[:2], radius, shape=shape)
+            pattern[rr, cc] = inten
+    if num_electrons is not None:
+        total = np.sum(spot_intens) * radius**2 * np.pi
+        pattern = np.random.poisson((pattern / total) * num_electrons)
+    return np.divide(pattern, np.max(pattern))
+
+
+def simulated_strain(
+    navigation_shape=(32, 32),
+    signal_shape=(512, 512),
+    disk_radius=20,
+    num_electrons=1e5,
+    strain_matrix=None,
+    lazy=False,
+):
+    if strain_matrix is None:
+        strain_matrix = np.array([[0.1, 0.05, 0], [0.15, 0.2, 0], [0, 0, 1]])
+    p = si_phase()
+    gen = SimulationGenerator()
+    rotations = Rotation.from_euler(
+        [
+            [0, 0, 0],
+        ],
+        degrees=True,
+    )
+    sim = gen.calculate_diffraction2d(
+        phase=p, rotation=rotations, reciprocal_radius=1.5, max_excitation_error=0.1
+    )
+
+    precip = np.zeros(navigation_shape, dtype=float)
+    r = navigation_shape[0] // 2
+    c = navigation_shape[1] // 2
+    r_rad = int(np.round(r * 0.5))
+    c_rad = int(np.round(c * 0.7))
+    rr, cc = ellipse(r, c, r_radius=r_rad, c_radius=c_rad)
+    precip[rr, cc] = 1
+
+    gaussian_filter(precip, sigma=3, output=precip)
+
+    data = np.empty(precip.shape + signal_shape)
+
+    for ind in np.ndindex(precip.shape):
+        t = np.eye(3) + strain_matrix * precip[ind]
+        data[ind] = create_diffraction_pattern(
+            sim,
+            shape=signal_shape,
+            radius=disk_radius,
+            num_electrons=num_electrons,
+            transformation_matrix=t,
+        )
+
+    strained = Diffraction2D(data)
+    strained.axes_manager.signal_axes[0].name = "kx"
+    strained.axes_manager.signal_axes[1].name = "kx"
+    strained.axes_manager.signal_axes[0].units = r"$\AA^{-1}$"
+    strained.axes_manager.signal_axes[1].units = r"$\AA^{-1}$"
+    strained.axes_manager.signal_axes[0].scale = 0.01
+    strained.axes_manager.signal_axes[1].scale = 0.01
+
+    strained.axes_manager.navigation_axes[0].name = "x"
+    strained.axes_manager.navigation_axes[1].name = "y"
+    strained.axes_manager.navigation_axes[0].units = "nm"
+    strained.axes_manager.navigation_axes[1].units = "nm"
```

Comment:
```suggestion
    strained.axes_manager.signal_axes.set(name=("x", "y"), units="nm")
```
Would it be worth setting a field of view and calculating the scale using the navigation shape?